packages/nc-gui/utils/parsers/CSVTemplateAdapter.ts

```diff
@@ -81,9 +81,16 @@ export default class CSVTemplateAdapter {
   }
 
   detectInitialUidt(v: string) {
+    const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
+    const urlRegex = /^(ftp|http|https):\/\/[^ "]+$/
+    const phoneRegex = /\b(?:\+?\d{1,3}(?:[\s-]+\(\d+\))?)?[\s-]*(?:\d[\s-]*){6,14}\b/
```

Comment:
This will treat any string containing a phone number as Phonenumber type.

For example:

```js
/\b(?:\+?\d{1,3}(?:[\s-]+\(\d+\))?)?[\s-]*(?:\d[\s-]*){6,14}\b/.test('abc +911234567890 abc')
```

Reply:
Replace word boundary with start and end anchors

```js
/^(?:\+?\d{1,3}(?:[\s-]+\(\d+\))?)?[\s-]*(?:\d[\s-]*){6,14}$/
```